package/MDAnalysis/analysis/Aggregation_size.py

```diff
@@ -0,0 +1,370 @@
+        """
+        self.results = np.zeros((self.n_frames, 5))
+        self.d = [0] * self._molnum
+        i = 0
+        for it in reversed(range(1, self._molnum+1)):
+            self.d[i] = np.zeros(it * 1).reshape(it, 1)
+            i += 1
+        self.size_num = [0] * self.n_frames
+
+    def aggregation(self, k, tss, counter, m, cut_off, connected):
+        """
+        This is a subfunction that enables the code to find all
+        connected molecules in a cluster.
+        k is a connected molecule to the main molecule in the first level.
+        self._frame_index is a time frame.
+        counter is a step.
+        m provides a suitable off_set to match the number of molecules.
+        Limit is the cut_off provided by the user.
+        """
+        bool_sector = self.d[k] < cut_off
+        Coordinates = self.d[k][bool_sector]
+        # This "if" will check if there is any uninvestigated molecule left
+        if len(Coordinates) != 0:
+            index1 = np.where(bool_sector)
+            listOfCoordinates = list(zip(index1[0], index1[1]))
+            connect = ([j[0] for j in listOfCoordinates]
+                       + connected[m] + 1)
+            Connected_NLevel = np.array(connect)
+        else:
+            Connected_NLevel = []
+        return Connected_NLevel
+
+    def _single_frame(self):
+        """
+        1- Import the needed module
+        2- Calculate the distance between molecules based on the intended
+        criteria
+        3- Find the aggregates by checking only one time each using the
+        "aggregate" subfunction (to be efficient)
+        4- Check and merge two aggregates as one aggregate if the distance
+        between any element of them is less than cut_off
+        5- Eliminate monomers (optional)
+        6- measure average gyration radius (optional)
+        7- measure the number of aggregates with different size
+        """
+        aggregate = [0] * self._molnum
+        if self._type == "Closest":
+            self.closest_atom(self.d)
+        elif self._type == "COM":
+            self.center_of_mass(self.d)
+        elif self._type == "Specific":
+            self.specific_atoms(self.d)
+        else:
+            raise ValueError("enter valid Type")
+        # P_Level collact all the investigated nodes in past levels
+        P_Level = []
+        # Group will collect the molecule first investigating molecule
+        # len(Group) will be the number of clusters
+        Group = []
+        # this for will go through all molecules to
+        # check their connection with others
+        for counter in range(1, self._molnum):
+            # aggergate parameter include the molecules in each cluster
+            aggregate[counter-1] = [counter]
+            # this "if" will restrict double calculation of the same molecule
+            # to increase efficincy of code
+            if counter not in P_Level:
```

Comment:
```suggestion
            if counter not in P_Level:
                continue
```
Make the loop more readable with early continue.

Then de-dent the rest of the code.

Reply:
I think @orbeckst meant ``if counter in P_level: continue`` 😄

Reply:
I think @orbeckst means:

```suggestion
            if counter in P_Level:
                continue
```

The idea is to abort the iteration early so the bulk can be one indentation level to the left.  The further the code is indented, the more difficult it becomes to read.